src/buildercore/terraform.py

```diff
@@ -35,3 +40,33 @@ def render(context):
         },
     }
     return json.dumps(tf_file)
+
+def init(stackname):
+    working_dir = join(TERRAFORM_DIR, stackname) # ll: ./.cfn/terraform/project--prod/
+    t = Terraform(working_dir=working_dir)
+    with open('%s/backend.tf' % working_dir, 'w') as fp:
+        fp.write(json.dumps({
+            'terraform': {
+                'backend': {
+                    's3': {
+                        'bucket': BUILDER_BUCKET,
+                        'key': 'terraform/%s.tfstate' % stackname,
+                        'region': BUILDER_REGION,
+                    },
+                },
+            },
+        }))
+    t.init(input=False, capture_output=False, raise_on_error=True)
+    return t
+
+@only_if('fastly')
+def update(stackname, context):
+    ensure('FASTLY_API_KEY' in os.environ, "a FASTLY_API_KEY environment variable is required to provision Fastly resources", ConfigurationError)
+    t = init(stackname)
+    t.apply(input=False, capture_output=False, raise_on_error=True)
+
+@only_if('fastly')
+def destroy(stackname, context):
+    t = init(stackname)
```

Comment:
By default single char variable names should be flagged as `bad-names` by `pylint`. Can't see a `good-names` override in the `.pylintrc` or `.lint.sh`, it must be disabling the the rule somewhere 🤔 .

Would advise adhering to single chars being bad names especially if there are no type hints present.

Reply:
will rename to `terraform` which doesn't shadow anything in this module

Reply:
To get it detected, I had to add `invalid-name` https://github.com/elifesciences/builder/blob/master/.lint.sh#L29
```
************* Module buildercore.terraform
C: 46, 4: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
```
That however finds too many occurrences in other files, so should be changed separately.